<?xml version="1.0"?>

<launch>
  <arg name="use_stability_assistance" default="$(optenv HECTOR_USE_STABILITY_ASSISTANCE false)"/>
  <arg name="cmd_vel_raw_topic" unless="$(arg use_stability_assistance)"  value="/cmd_vel_raw"/>
  <arg name="cmd_vel_raw_topic" if="$(arg use_stability_assistance)"      value="/cmd_vel_unlimited"/>
  
  <node pkg="vehicle_controller" type="controller_node" name="vehicle_controller" output="screen" respawn="true">
    <param name="frame_id" value="world" />
    <param name="carrot_controller_params/carrot_distance" value="0.2"/>
    <param name="y_symmetry" value="false" />
    
    <param name="max_controller_speed" value="1.0" />
    <param name="max_unlimited_speed" value="1.5" />
    <param name="max_unlimited_angular_rate" value="3.0" />
    <param name="max_controller_angular_rate" value="1.5" />
    <param name="wheel_separation" value="0.49882" />
    
    <!-- Carrot (0), DAF (1), Ackermann Pure Pursuit (2), Differential Pure Pursuit (3), LQR (4) -->
    <param name="controller_type" value="4"/>
    <param name="vehicle_control_type" value="differential_steering"/>

    <param name="goal_position_tolerance" value="0.1" />
    <param name="goal_angle_tolerance" value="0.05" />
    <param name="pd_params/speed" value="0.3"/>
    <param name="pd_params/angle_p_gain" value="1.0"/>
    <param name="pd_params/angle_d_gain" value="0.0"/>
    
    <param name="check_stuck" value="true" /> 
    
    <remap from="/state" to="/odom" />
    <remap from="/cmd_vel_raw" to="$(arg cmd_vel_raw_topic)" />
  </node>
  
</launch>

